sklearn/utils/_mask.py

```diff
@@ -1,21 +1,40 @@
 import numpy as np
+from scipy import sparse
 
 from . import is_scalar_nan
 from .fixes import _object_dtype_isnan
 
 
-def _get_mask(X, value_to_mask):
+def _get_mask(X, value_to_mask, reconstruct_sparse=False):
     """Compute the boolean mask X == missing_values."""
+    X_ = X.copy()
+    if reconstruct_sparse:
+        # if sparse.issparse(X):
+        X_ = X.data
+
     if is_scalar_nan(value_to_mask):
-        if X.dtype.kind == "f":
-            return np.isnan(X)
+        if X_.dtype.kind == "f":
+            Xt = np.isnan(X_)
         elif X.dtype.kind in ("i", "u"):
             # can't have NaNs in integer array.
-            return np.zeros(X.shape, dtype=bool)
+            Xt = np.zeros(X_.shape, dtype=bool)
         else:
             # np.isnan does not work on object dtypes.
-            return _object_dtype_isnan(X)
+            Xt = _object_dtype_isnan(X_)
     else:
         # X == value_to_mask with object dtypes does not always perform
         # element-wise for old versions of numpy
-        return np.equal(X, value_to_mask)
+        Xt = np.equal(X_, value_to_mask)
+
+    if not reconstruct_sparse:
+        return Xt
+
+    else:
+        sparse_constructor = (sparse.csr_matrix
+                              if X.format == 'csr'
+                              else sparse.csc_matrix)
```

Comment:
this is making assumptions not given in the docstring of the function. can't format be csc?

Reply:
I'm kind of confused, but as per the implementation if the format is not `csr` we will conver it to `csc`.